Retarus/retarus_modules/consumer.py

```diff
@@ -76,13 +77,18 @@ def on_close(self, *_):
     def run(self):
         """Start the websocket thread then wait for the stop event to be set and close the websocket when it happens"""
         self.log(message=f"Connection to stream {self.configuration.ws_url}", level="info")
+        while self.is_running:
+            self.websocket = self.create_websocket()
```

Comment:
The websocket seems to be used only here

```suggestion
            websocket = self.create_websocket()
```